### PR TITLE
gsoc begins blog post

### PR DESCRIPTION
**Pull request by DevinR528**

### CODE REVIEW

File: content/news/2020-06-01-gsoc-begins-2020.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
++++
+title = "Google Summer of Code begins"
```
Comment (jplatte) [inline content/news/2020-06-01-gsoc-begins-2020.md]: I don't think we need two titles here. Just set this to `Ruma gets in on Google Summer of Code` and remove the heading below.

File: content/news/2020-06-01-gsoc-begins-2020.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+date = 2020-06-01
+extra.author = "Devin Ragotzy"
++++
+
+## Ruma gets in on Google Summer of Code
+
+Thanks to GSoC and the continued help of Jonas Platte, Ruma's GSoC mentors, there is a new contributor... me! My name is Devin Ragotzy, a University student from Michigan. I was chosen to help contribute to the macro code of ruma-events and ruma-api. Before the official coding period began on June 1st, a large breaking change started taking shape for the upcoming release of ruma-events [0.22.0](https://github.com/ruma/ruma-events/issues/82). Jonas, I, and others have been hard at work, replacing the previous event declaration macro with a set of derive macros and generic event types, as well as getting rid of the boilerplate required for event enums (the types currently found under ruma_events::collections). The new types will be able to express JSON events from a Matrix server with more accuracy.
```
Comment (jplatte) [inline content/news/2020-06-01-gsoc-begins-2020.md]:
```suggestion
Thanks to GSoC and the continued help of Jonas Platte, Ruma's GSoC mentor, there is a new contributor... me! My name is Devin Ragotzy, a University student from Michigan. I was chosen to help contribute to the macro code of ruma-events and ruma-api. Before the official coding period began on June 1st, a large breaking change started taking shape for the upcoming release of ruma-events [0.22.0](https://github.com/ruma/ruma-events/issues/82). Jonas, I, and others have been hard at work, replacing the previous event declaration macro with a set of derive macros and generic event types, as well as getting rid of the boilerplate required for event enums (the types currently found under `ruma_events::collections`). The new types will be able to express JSON events from a Matrix server with more accuracy.
```

**Review by jplatte — COMMENTED** [review]
Just had a thought: Should we also touch on the planned work on ruma-api-macros or should we leave that to another blog post that you write once you've actually started working on that?

**Comment by DevinR528** [comment]
My thinking was to keep it to what I had done, but maybe a link to/including some of the project abstract?

**Comment by jplatte** [comment]
You mean [the proposal on the GSoC website](https://summerofcode.withgoogle.com/projects/#4756651216732160)? Or is there something else we can link to?

**Comment by DevinR528** [comment]
The GSoC website.

File: content/news/2020-06-01-gsoc-begins-2020.md [hunk 3]
```diff
@@ -5,3 +5,12 @@ extra.author = "Devin Ragotzy"
...
+
+## Planned work for ruma-api
+
+Once the macros in ruma-events are implemented, I will switch gears to work on ruma-api.
+- First, add a few tests to ensure the changes I make do not break important functionality
+  - Test macro code using `macrotest` and/or `trybuild` which test errors or expansions respectively
+- Allow more code sharing between `Request` and `Response` processing inside the macro
+- Refactor `ruma_api!` breaking up the large block of code in `Api::to_tokens`
```
Comment (jplatte) [inline content/news/2020-06-01-gsoc-begins-2020.md]:
```suggestion
- Refactor `ruma_api!`, breaking up the large block of code in `Api::to_tokens`
```

File: content/news/2020-06-01-gsoc-begins-2020.md [hunk 4]
```diff
@@ -5,3 +5,12 @@ extra.author = "Devin Ragotzy"
 +++
 
 Thanks to GSoC and the continued help of Jonas Platte, Ruma's GSoC mentor, there is a new contributor... me! My name is Devin Ragotzy, a University student from Michigan. I was chosen to help contribute to the macro code of ruma-events and ruma-api. Before the official coding period began on June 1st, a large breaking change started taking shape for the upcoming release of ruma-events [0.22.0](https://github.com/ruma/ruma-events/issues/82). Jonas, I, and others have been hard at work, replacing the previous event declaration macro with a set of derive macros and generic event types, as well as getting rid of the boilerplate required for event enums (the types currently found under `ruma_events::collections`). The new types will be able to express JSON events from a Matrix server with more accuracy.
+
+## Planned work for ruma-api
+
+Once the macros in ruma-events are implemented, I will switch gears to work on ruma-api.
+- First, add a few tests to ensure the changes I make do not break important functionality
```
Comment (jplatte) [inline content/news/2020-06-01-gsoc-begins-2020.md]: There should be an empty line between a paragraph and a list.

```suggestion
Once the macros in ruma-events are implemented, I will switch gears to work on ruma-api.

- First, add a few tests to ensure the changes I make do not break important functionality
```

**Review by jplatte — COMMENTED** [review]
The new section looks good! If we have a sub-heading for ruma-api though, we should probably have one for ruma-events. Can you split the first paragraph before

> Before the official coding period began

and add a sub-heading for that part?

**Review by jplatte — APPROVED** [review]

**Pull request merged**